native/game_backend/src/map.rs

```diff
@@ -7,13 +7,13 @@ use crate::player::Player;
 
 #[derive(NifMap, Clone, Copy)]
 pub struct Position {
-    pub x: i64,
-    pub y: i64,
+    pub x: f32,
+    pub y: f32,
 }
 
 pub fn hit_boxes_collide(center1: &Position, center2: &Position, size1: u64, size2: u64) -> bool {
-    let squared_x = (center1.x - center2.x).pow(2) as f64;
-    let squared_y = (center1.y - center2.y).pow(2) as f64;
+    let squared_x = ((center1.x - center2.x) as f64).powi(2);
```

Comment:
Should this also change to f32?

Reply:
Since this the exponentiation of an f32, it can turn into a number bigger than f32::max, so the f64 gives it a bit of head room.